packages/juce-blueprint/src/lib/SyntheticEvents.ts

```diff
@@ -0,0 +1,69 @@
+export class SyntheticEvent {
+  public  bubbles: boolean;
+  public  defaultPrevented: boolean;
+  private _internal: any;
+
+  constructor(props: any) {
+    this.bubbles = true;
+    this.defaultPrevented = false;
+
+    this._internal = props;
+  }
+
+  stopPropagation(): void {
+    this.bubbles = false;
+  }
+
+  preventDefault(): void {
+    this.defaultPrevented = true;
+  }
+}
+
+export class SyntheticMouseEvent extends SyntheticEvent {
+  public x: number; 
+  public y: number;
+  public clientX: number;
+  public clientY: number;
+  public screenX: number;
+  public screenY: number;
+
+  constructor(props: any) {
+    super(props);
+
+    this.x = this.clientX = props.x;
+    this.y = this.clientY = props.y;
+    this.screenX = props.screenX;
+    this.screenY = props.screenY;
+  }
+}
+
+export class SyntheticKeyboardEvent extends SyntheticEvent {
+  public keyCode: number;
+  public key: string;
+
+  constructor(props: any) {
+    super(props);
+
+    this.keyCode = props.keyCode;
+    this.key = props.key;
+  }
+}
+
+export default {
```

Comment:
Does this feel like the write approach for SyntheticEvents? Add the helpers to the `default` export?

Reply:
Yea, these should be implementaion details, only used when we're trying to decide what to bubble up the tree. Let's hide them from the user

Reply:
Heh. So happy as they are for now or further hiding? Can move helpers file for internal import of desired, I'm not sure what TS/JS module beat practice dictates here.

Reply:
Yea, we can definitely leave them here for now. In a future PR, let's pull these out into a `utils/EventHelpers.js` or some such file, then any of our internal code that needs it can just `import { isMouseEvent } from "utils/EventHelpers"` or so, and that way these helpers won't end up in the user-facing package because our barrel file won't import or re-export any of the helpers, and any of our other code that relies on them will be importing, but not exporting them.

Reply:
Yeah totally. I'll fire up an issue for that as a minor refactor.